multi-node/emulate-multi-node.md

```diff
@@ -7,191 +7,194 @@ We use the `deepspeed` launcher here. There is no need to actually use any of th
+         printflock(f"device compute capabilities={torch.cuda.get_device_capability()}")
+         printflock(f"pytorch compute capabilities={torch.cuda.get_arch_list()}")
+ 
+ except Exception:
+     printflock(f"{gpu} is broken")
+     raise
+ ```
+ 
+ Run:
+ 
+```bash
+ $ deepspeed -H hostfile test2.py
+ [2022-09-08 12:07:09,336] [INFO] [runner.py:415:main] Using IP address of 192.168.0.17 for node worker-0
+ [2022-09-08 12:07:09,337] [INFO] [multinode_runner.py:65:get_cmd] Running on the following workers: worker-0,worker-1
+ [2022-09-08 12:07:09,337] [INFO] [runner.py:504:main] cmd = pdsh -S -f 1024 -w worker-0,worker-1 export PYTHONPATH=/mnt/nvme0/code/huggingface/multi-node-emulate-ds;  cd /mnt/nvme0/code/huggingface/multi-node-emulate-ds; /home/stas/anaconda3/envs/py38-pt112/bin/python -u -m deepspeed.launcher.launch --world_info=eyJ3b3JrZXItMCI6IFswXSwgIndvcmtlci0xIjogWzBdfQ== --node_rank=%n --master_addr=192.168.0.17 --master_port=29500 test2.py
+ worker-0: [2022-09-08 12:07:10,635] [INFO] [launch.py:136:main] WORLD INFO DICT: {'worker-0': [0], 'worker-1': [0]}
+ worker-0: [2022-09-08 12:07:10,635] [INFO] [launch.py:142:main] nnodes=2, num_local_procs=1, node_rank=0
+ worker-0: [2022-09-08 12:07:10,635] [INFO] [launch.py:155:main] global_rank_mapping=defaultdict(<class 'list'>, {'worker-0': [0], 'worker-1': [1]})
+ worker-0: [2022-09-08 12:07:10,635] [INFO] [launch.py:156:main] dist_world_size=2
+ worker-0: [2022-09-08 12:07:10,635] [INFO] [launch.py:158:main] Setting CUDA_VISIBLE_DEVICES=0
+ worker-1: [2022-09-08 12:07:10,635] [INFO] [launch.py:136:main] WORLD INFO DICT: {'worker-0': [0], 'worker-1': [0]}
+ worker-1: [2022-09-08 12:07:10,635] [INFO] [launch.py:142:main] nnodes=2, num_local_procs=1, node_rank=1
+ worker-1: [2022-09-08 12:07:10,635] [INFO] [launch.py:155:main] global_rank_mapping=defaultdict(<class 'list'>, {'worker-0': [0], 'worker-1': [1]})
+ worker-1: [2022-09-08 12:07:10,635] [INFO] [launch.py:156:main] dist_world_size=2
+ worker-1: [2022-09-08 12:07:10,635] [INFO] [launch.py:158:main] Setting CUDA_VISIBLE_DEVICES=0
+ worker-0: dist.get_rank()=0, local_rank=0
+ worker-1: dist.get_rank()=1, local_rank=0
+ worker-0: [hope-0] is OK (global rank: 0/2)
+ worker-1: [hope-0] is OK (global rank: 1/2)
+ worker-0: pt=1.12.1+cu116, cuda=11.6, nccl=(2, 10, 3)
+ worker-0: device compute capabilities=(8, 0)
+ worker-0: pytorch compute capabilities=['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80', 'sm_86']
+ worker-1: [2022-09-08 12:07:13,642] [INFO] [launch.py:318:main] Process 576485 exits successfully.
+ worker-0: [2022-09-08 12:07:13,642] [INFO] [launch.py:318:main] Process 576484 exits successfully.
+ ```
+ 
+ Voila, missing accomplished.
+ 
+ We tested that the NCCL collectives work, but they use local NVLink/PCIe and not the IB/ETH connections like in real multi-node, so it may or may not be good enough for testing depending on what needs to be tested.
```

Comment:
this needs to be fixed as well please - you added a leading whitespace to this whole section

Reply:
Fixed in [78f30a](https://github.com/saforem2/toolbox-fork/commit/78f30afa1f84b3c960e41f336beb1cdba14577ef

Reply:
that didn't work. This time you changed the original indentation.

Reply:
oops. should be fixed now in [247ca3](https://github.com/saforem2/toolbox-fork/commit/247ca3dd5d5f48f21ec2ae73714f52386023ed32)